NDTensors/src/tensorstorage/tensorstorage.jl

```diff
@@ -1,8 +1,11 @@
+using .TypeParameterAccessors: TypeParameterAccessors, parameter, parenttype
+
 abstract type TensorStorage{ElT} <: AbstractVector{ElT} end
 
 data(S::TensorStorage) = S.data
 
-datatype(::Type{<:TensorStorage}) = error("Not implemented")
+datatype(type::Type{<:TensorStorage}) = parenttype(type)
+TypeParameterAccessors.parenttype(type::Type{<:TensorStorage}) = parameter(type, parenttype)
```

Comment:
Can't this be a generic function:
```julia
TypeParameterAccessors.parenttype(type::Type) = parameter(type, parenttype)
```
for any type?

Reply:
So I added this to `TypeParameterAccessors` and went with the `Self()` definition for now but this definitely can be changed in the future.